### PR TITLE
dev/core#3961 - Move component Api4 files to component extensions

### PR DESCRIPTION
**Pull request by colemanw**
Overview
----------------------------------------
Following on #26036 this is a very small step toward moving component code into extensions: this moves the APIv4 classes.
See [dev/core#3961 proposal to create a stub extension for all components](https://lab.civicrm.org/dev/core/-/issues/3961).

Before
----------------------------------------
Component api files in core.

After
----------------------------------------
Component api files in component extensions

Technical Details
----------------------------------------
This moves the code belonging to components into their extension directories. As before, these apis will only be available if the component is enabled, but now there is no extra code needed to make that happen.

### CODE REVIEW

**Comment by totten** [comment]
Related discussion: https://chat.civicrm.org/civicrm/pl/159e3dzuw7yuxj1zxqjg6zx5pw

> As before, these apis will only be available if the component is enabled, but now there is no extra code needed to make that happen.

Right, I see what you're saying. That's the linchpin.

Do any more edges come to mind if you think about:

* Metadata bits like `::permissions()` or `::getFields()`?
* For these APIs, was it always necessary to have the component enabled? I thought that had changed somewhere in the past year?

--

`FiveSixtyOne::convertMappingFieldsToApi4StyleNames()` fails when it calls `Civi\API\Contribution::getFields()` during an upgrade. 

I bet this is failing because of the order-of-operations.  At the time when `FiveSixtyOne` executes, the `civicrm_extension[full_name=civi_contribute]` is not installed. So naturally `FiveSixtyOne` shouldn't have access to resources from `ext/civi_contribute/`. At best, maybe those resources become accessible after `FiveSixtyThree`.

Perhaps the issue would be moot if `FiveSixtyOne` used low-dependency code - maybe:

```php
$fieldMap += CRM_Core_DAO::executeQuery('
  SELECT CONCAT("custom_", fld.id) AS old, CONCAT(grp.name,".",fld.name) AS new 
  FROM civicrm_custom_field fld...
')->fetchMap('old', 'new');
```

**Comment by colemanw** [comment]
Thanks @totten I've updated the upgrader per your suggestion. To your other questions:
- I can't think of any gotchas with metadata bits
- APIv4 has excluded entities from disabled components since d31fb4e3103752d7a0d72f453644de99f726f8da

File: tests/phpunit/api/v4/Entity/ConformanceTest.php [hunk 1]
```diff
@@ -85,10 +85,6 @@ public function tearDown(): void {
    * @throws \CRM_Core_Exception
    */
   public function getEntitiesHitech(): array {
-    // Ensure all components are enabled so their entities show up
-    foreach (array_keys(\CRM_Core_Component::getComponents()) as $component) {
-      \CRM_Core_BAO_ConfigSetting::enableComponent($component);
-    }
```
Comment (colemanw) [inline tests/phpunit/api/v4/Entity/ConformanceTest.php:91]: This was redundant with the `setUp` function in this class which calls `enableAllComponents()`

**Comment by totten** [comment]
Yup, we've just started the cycle for 5.64.alpha - seems like a good time to get this in. Merge on pass.

**Pull request merged**